### PR TITLE
Issue87 file archive delete fix

### LINKED ISSUE #87: Failing delete
```
$> curl --data-binary "@precipitation_amount_st_92350.xml" localhost:5000/v1/insert
$> curl --data-binary "@precipitation_amount_st_92350.xml" localhost:5000/v1/delete/ee6fb8de-8ebd-4df6-95dd-83a44d21dfc7
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>
```

Output from the server:
```
ERROR    Exception on /v1/delete/ee6fb8de-8ebd-4df6-95dd-83a44d21dfc7 [POST]
Traceback (most recent call last):
  File "/home/morten/miniconda3/envs/dmci/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/morten/miniconda3/envs/dmci/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/morten/miniconda3/envs/dmci/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/morten/miniconda3/envs/dmci/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/morten/miniconda3/envs/dmci/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/morten/miniconda3/envs/dmci/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/morten/metno_repos/discovery-metadata-catalog-ingestor/dmci/api/app.py", line 73, in post_delete
    err = self._distributor_wrapper(worker)
  File "/home/morten/metno_repos/discovery-metadata-catalog-ingestor/dmci/api/app.py", line 164, in _distributor_wrapper
    status, valid, _, failed, skipped, failed_msg = worker.distribute()
  File "/home/morten/metno_repos/discovery-metadata-catalog-ingestor/dmci/api/worker.py", line 123, in distribute
    obj = self.CALL_MAP[dist](
TypeError: dmci.distributors.file_dist.FileDist() got multiple values for keyword argument 'metadata_id'
127.0.0.1 - - [04/May/2022 14:24:11] "POST /v1/delete/ee6fb8de-8ebd-4df6-95dd-83a44d21dfc7 HTTP/1.1" 500 -
```

### LINKED ISSUE #88: DMCI insert should be changed to create
I.e., `curl --data-binary "@<PATH_TO_MMD_FILE>" localhost:5000/v1/insert` should be `curl --data-binary "@<PATH_TO_MMD_FILE>" localhost:5000/v1/create`

NOTE: v0.3 can have both `insert` and `create` but only `create` must be in v1.0 onwards.

### PR DESCRIPTION
**Pull request by johtoblan**
**Summary**:

- Adds alias for insert command: create
- Updates readme to add this
- Fixes an issue with keyword argument dictionary being unpacked causing multiple instances of some keywords
- Delete_from_archive in file_distributor does not use uninstanced _file_metadata_id
- Enforce metadata_id to be of type UUID 

**Related issue**:

Closes #87 
Partly? Closes #88 

**Suggested reviewer(s)**:

@mortenwh 

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,9 +65,13 @@ python dmci_start_api.py
...
 ```bash
-curl --data-binary "@<PATH_TO_MMD_FILE>" localhost:5000/v1/insert
+# Validate, insert, create and update call signature
+curl --data-binary "@<PATH_TO_MMD_FILE>" localhost:5000/v1/validate
+# Delete works differently
+curl -X POST localhost:5000/v1/delete/<UUID_OF_FILE_TO_DELETE>
+
 ```
-insert is currently the only implemented command. Other commands will return 404 Not Found until implementation.
+Available commands are: validate, insert or create, update, and delete.
```
Comment (mortenwh) [inline README.md:74]:
```suggestion
Available commands are: validate, insert or create, update, and delete. Note that insert and create is the same - insert will be removed in the next major version (1.0).
```

File: dmci/api/worker.py [hunk 2]
```diff
@@ -119,13 +119,11 @@ def distribute(self):
             if dist not in self.CALL_MAP:
                 skipped.append(dist)
                 continue
-
             obj = self.CALL_MAP[dist](
                 self._dist_cmd,
                 xml_file=self._dist_xml_file,
                 metadata_id=self._dist_metadata_id,
                 worker=self,
```
Comment (mortenwh) [inline dmci/api/worker.py:126]:
```suggestion
                worker=self
```

**Review by mortenwh — COMMENTED** [review]
Only minor comments. You mention that this PR partly closes #88. I removed the automatically generated connection to #88 just in case. Can you double check? It seems fine to me.

**Comment by mortenwh** [comment]
btw - I guess #88 can be closed if we make a new issue "Remove insert command" and mark it as a v1.0 release (or milestone)..

**Comment by johtoblan** [comment]
> Only minor comments. You mention that this PR partly closes #88. I removed the automatically generated connection to #88 just in case. Can you double check? It seems fine to me.

Yes, partly since not removing insert.

> btw - I guess #88 can be closed if we make a new issue "Remove insert command" and mark it as a v1.0 release (or milestone)..


Yes.

**Review by mortenwh — APPROVED** [review]

**Pull request merged**